PyPoE/poe/file/ggpk.py

```diff
@@ -382,13 +382,14 @@ def extract_to(self, directory, name=None):
     def read(self, ggpkfile):
         self._name_length = struct.unpack('<i', ggpkfile.read(4))[0]
         self.hash = int.from_bytes(ggpkfile.read(32), 'big')
-        # UTF-16 2-byte width
-        self._name = ggpkfile.read(2 * (self._name_length - 1)).decode('UTF-16')
+        # UTF-16/32 characters
+        wchar_width = self._container.wchar_width
+        self._name = ggpkfile.read(wchar_width * (self._name_length - 1)).decode(self._container.wchar_encoding)
         # Null Termination
-        ggpkfile.seek(2, os.SEEK_CUR)
+        ggpkfile.seek(wchar_width, os.SEEK_CUR)
         self.data_start = ggpkfile.tell()
         # Length 4B - Tag 4B - STRLen 4B - Hash 32B + STR ?B  
-        self.data_length = self.length - 44 - self._name_length * 2
+        self.data_length = self.length - 44 - self._name_length * wchar_width
```

Comment:
This is not really about your change, but in general I've been seeing lots of magic numbers in the repo, while the comments above explain them, I wonder if this is the best way going forward as opposed to things like enum members?